pages/events/index.vue

```diff
@@ -108,11 +104,39 @@ type Event = {
     }
   },
 
-  async asyncData () {
-    const eventsModule = await import('~/content/events/events-previews.json')
-    const events: Array<Event> = eventsModule.default || []
+  computed: {
+    ...mapGetters([
+      'filteredEvents',
+      'typeFilters',
+      'locationFilters'
+    ])
+  },
+
+  methods: {
+    ...mapActions({
+      fetchEvents: 'fetchEvents'
+    }),
+    isFilterChecked (filter, filterValue) {
+      const typeFilters = (this as any).typeFilters
+      const locationFilters = (this as any).locationFilters
+
+      return filter === 'locationFilters'
+        ? locationFilters.includes(filterValue)
+        : typeFilters.includes(filterValue)
+    },
+    updateFilter (filter, filterValue, selectFilter) {
```

Comment:
Can we reflect the fact that `selectFilter` act as a flag renaming it to `isSelected` or something similar?